### PR TITLE
Swap old JS client for an Angular one

### PR DESCRIPTION
**Pull request by tbouron**
This uses:
* angular 1.4.8
* bootstrap
* font-awesome

It also introduces a new way of building / developing the client. There are 2 cases:
* **for development purpose**: use gulp that will compile and copy all required resources into `target/app` and serve them through a local web server at `http://localhost:8000`
* **for building / release purpose**: simply run the classic `mvn clean install` which will call the `gulp` under the hood to build the resources then maven will package them as a web fragment JAR for the `dist` module.

More info in the `client/README.md` file.

### CODE REVIEW

File: client/gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    connect.server({
+        root: targetRoot,
+        port: 8000,
+        livereload: true
+    });
+});
+
+// Build the app
+gulp.task('build', ['js', 'less', 'img', 'partial', 'index'], function() {
+    console.log('✔ Build done!');
```
Comment (m4rkmckenna) [inline client/gulpfile.js]: personal preference to use `gulp-util` 
```
gutil.log('stuff happened', 'Really it did', gutil.colors.magenta('123'));
gutil.beep();
```

**Comment by m4rkmckenna** [comment]
LGTM

love what you've done with gulp

**Comment by tbouron** [comment]
@m4rkmckenna Thanks! Gulp is really great and more powerful than Grunt IMO. I'll address your comment and push again.

**Comment by tbouron** [comment]
@m4rkmckenna Comment addressed

**Pull request merged**